File: app/src/main/java/com/arjanvlek/oxygenupdater/download/UpdateDownloadListener.java
package com.arjanvlek.oxygenupdater.download;

public interface UpdateDownloadListener {

	void onInitialStatusUpdate();

	void onDownloadStarted();

	void onDownloadProgressUpdate(DownloadProgressData downloadProgressData);

	void onDownloadPaused(boolean queued, DownloadProgressData downloadProgressData);

	void onDownloadComplete();

	void onDownloadCancelled();

	void onDownloadError(boolean isInternalError, boolean isStorageSpaceError, boolean isServerError);

	void onVerifyStarted();

	void onVerifyError();

	void onVerifyComplete(boolean launchInstallation);

}

File: app/src/main/java/com/arjanvlek/oxygenupdater/download/UpdateDownloadListener.kt
package com.arjanvlek.oxygenupdater.download

import com.arjanvlek.oxygenupdater.models.DownloadProgressData

interface UpdateDownloadListener {
    fun onInitialStatusUpdate()
    fun onDownloadStarted()
    fun onDownloadProgressUpdate(downloadProgressData: DownloadProgressData)
    fun onDownloadPaused(queued: Boolean, downloadProgressData: DownloadProgressData)
    fun onDownloadComplete()
    fun onDownloadCancelled()
    fun onDownloadError(isInternalError: Boolean, isStorageSpaceError: Boolean, isServerError: Boolean)
    fun onVerifyStarted()
    fun onVerifyError()
    fun onVerifyComplete(launchInstallation: Boolean)
}
